src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -1300,9 +1241,15 @@ private StudyCreationRequestEntity insertStudyCreationRequestEntity(UUID studyUu
         return studyCreationRequestRepository.save(studyCreationRequestEntity);
     }
 
-    public void updateLoadFlowParameters(UUID studyUuid, LoadFlowParametersEntity loadFlowParametersEntity) {
-        Optional<StudyEntity> studyEntity = studyRepository.findById(studyUuid);
-        studyEntity.ifPresent(studyEntity1 -> studyEntity1.setLoadFlowParameters(loadFlowParametersEntity));
+    public void createOrUpdateLoadFlowParameters(UUID studyUuid, String parameters) {
+        StudyEntity studyEntity = studyRepository.findById(studyUuid).orElseThrow(() -> new StudyException(STUDY_NOT_FOUND));
+        UUID loadFlowParametersUuid = studyEntity.getLoadFlowParametersUuid();
+        if (loadFlowParametersUuid == null) {
+            loadFlowParametersUuid = loadflowService.createLoadFlowParameters(parameters);
+            studyEntity.setLoadFlowParametersUuid(loadFlowParametersUuid);
```

Comment:
you never save this new entity and you are not in a transaction so it's gone into the void

Reply:
Which makes me think : this case is probably not tested. You should probably add one.

Reply:
fixed

Reply:
You did not add the test